### PR TITLE
Namespace fixes for - glTF Utilities based off of built in Unity Serialization

### PR DESCRIPTION
**Pull request by chrisfromwork**
Overview
---
This is a new pull request to address some namespace breaks related to https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3548

Changes
---
- Fixes: #3595 .

### CODE REVIEW

File: Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerVisualizationProfile.cs [hunk 1]
```diff
@@ -110,5 +110,47 @@ public GameObject GetControllerModelOverride(Type controllerType, Handedness han
...
+        /// </summary>
+        /// <param name="controllerType">The type of controller to query for</param>
+        /// <param name="hand">The specific hand assigned to the controller</param>
+        public SystemType GetControllerVisualizationTypeOverride(Type controllerType, Handedness hand)
+        {
+            for (int i = 0; i < controllerVisualizationSettings.Length; i++)
+            {
+                if (controllerVisualizationSettings[i].ControllerType != null &&
+                    controllerVisualizationSettings[i].ControllerType.Type == controllerType &&
+                    (controllerVisualizationSettings[i].Handedness == hand || controllerVisualizationSettings[i].Handedness == Handedness.Both))
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerVisualizationProfile.cs]: These conditions are shared between GetControllerVisualizationTypeOverride and GetControllerModelOverride and GetUseDefaultModelsOverride, and probably could be a little helper.

File: Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfConversions.cs [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+            bool mirrored = Vector3.Dot(calculatedZ, z) < 0.0f;
+
+            scale.x = x.magnitude * (mirrored ? -1.0f : 1.0f);
+            scale.y = y.magnitude;
+            scale.z = z.magnitude;
+
+            rotation = Quaternion.LookRotation(matrix.GetColumn(2), matrix.GetColumn(1));
+        }
+
+        internal static int[] GetIntArray(this GltfAccessor accessor, bool flipFaces = true)
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfConversions.cs]: >flipFaces [](start = 75, length = 9)

What is flipFaces?

File: Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+            {
+                Debug.LogError("Unsupported file name extension.");
+                return null;
+            }
+
+            gltfObject.Uri = uri;
+            int nameStart = uri.Replace("\\", "/").LastIndexOf("/", StringComparison.Ordinal) + 1;
+            int nameLength = uri.Length - nameStart;
+            gltfObject.Name = uri.Substring(nameStart, nameLength).Replace(isGlb ? ".glb" : ".gltf", string.Empty);
+
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs]: Might be possible to use a path API that does this:

[https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=netframework-4.7.2](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getfilenamewithoutextension?view=netframework-4.7.2)

It looks like it exists in the .NET versions we care about.

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityController.cs [hunk 4]
```diff
@@ -511,8 +525,131 @@ private void UpdateMenuData(InteractionSourceState interactionSourceState, Mixed
             }
         }
 
+        /// <summary>
+        /// Update the Controller Model
+        /// </summary>
+        /// <param name="interactionSourceState"></param>
+        private void UpdateControllerModel(InteractionSourceState interactionSourceState)
+        {
+            if (controllerModelInitialized ||
```
Comment (wiwei) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityController.cs]: >controllerModelInitialized [](start = 16, length = 26)

Once this value is set to true, it looks like UpdateControllerModel will never actually update anyting - is this just maybe misnamed (i.e. it's more like "ensure controller model"). 

Given the current naming I would expect the controller model to update in response to changes, but once this value is true, this entire statement will short circuit and nothing will happen.

File: Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfConversions.cs (hunk 2 above)
Comment (chrisfromwork) [reply Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfConversions.cs]: flip faces changes the triangle definitions/indices order. This changes the outward/normal direction which then results in changing which side of the mesh surface is drawn.

I think that the MRTK standard shader has some different flags that can override/disregard this behavior and draw both sides. But i would also guess this is a direct port from khronos's logic and should be kept

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityController.cs (hunk 4 above)
Comment (chrisfromwork) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealityController.cs]: agreed that this is confusing. will update to EnsureControllerModel

File: Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerVisualizationProfile.cs (hunk 1 above)
Comment (chrisfromwork) [reply Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerVisualizationProfile.cs]: changed to SettingContainsParameters

File: Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs (hunk 3 above)
Comment (chrisfromwork) [reply Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs]: seems to work. Removes some flags/seems good

**Pull request merged**